### PR TITLE
Add Gaussian Vertex distribution

### LINKED ISSUE #206: [Generation] Add Gaussian Vertex Smearing
An additional implementation of the VertexSmearingTool with a normal distribution is needed.

### PR DESCRIPTION
**Pull request by vvolkl**
Fixes #206.

Changes proposed in this pull-request:

- Adds an implementation of the VertexSmearingTool with a normal distribution
- Synchronizes some parameters with TkLayout Studies

### CODE REVIEW

File: Examples/options/simple_pythia.py [hunk 1]
```diff
@@ -16,15 +16,24 @@
...
+from Configurables import ConstPileUp, HepMCFileReader, GaussSmearVertex
+
+from FCCPileupScenarios import FCCPhase1PileupConf as pileupconf
+smeartool = GaussSmearVertex(
+     xVertexMin=pileupconf['xVertexMin'],
+     xVertexMax=pileupconf['xVertexMax'],
+     yVertexMin=pileupconf['yVertexMin'],
+     yVertexMax=pileupconf['yVertexMax'],
+     zVertexMin=pileupconf['zVertexMin'],
+     zVertexMax=pileupconf['zVertexMax'])
```
Comment (jlingema) [inline Examples/options/simple_pythia.py]: For a Gaussian I'd expect mean and spread / variance here, not min / max?

File: Generation/src/components/GaussSmearVertex.cpp [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+DECLARE_TOOL_FACTORY(GaussSmearVertex)
+
+/// Standard constructor, initializes variables
+GaussSmearVertex::GaussSmearVertex(const std::string& type, const std::string& name, const IInterface* parent)
+    : GaudiTool(type, name, parent) {
+  declareInterface<IVertexSmearingTool>(this);
+
+  declareProperty("xVertexSigma", m_xsig = 0.0 * Gaudi::Units::mm);
+  declareProperty("yVertexSigma", m_ysig = 0.0 * Gaudi::Units::mm);
+  declareProperty("zVertexSigma", m_zsig = 0.0 * Gaudi::Units::mm);
```
Comment (jlingema) [inline Generation/src/components/GaussSmearVertex.cpp]: Why are these also declared as Properties if they're computed below from the other properties? Also move them into the header.

**Comment by vvolkl** [comment]
Thanks Joschka. I Changed the tool properties to avoid confusion.

File: Generation/src/components/GaussSmearVertex.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  Gaudi::Property<double> m_xmean{this, "xVertexMean", 0.0 * Gaudi::Units::mm, "Mean of x coordinate"};
+  Gaudi::Property<double> m_ymean{this, "yVertexMean", 0.0 * Gaudi::Units::mm, "Mean of y coordinate"};
+  Gaudi::Property<double> m_zmean{this, "zVertexMean", 0.0 * Gaudi::Units::mm, "Mean of z coordinate"};
+  Gaudi::Property<double> m_tmean{this, "tVertexMean", 0.0 * Gaudi::Units::mm, "Mean of t coordinate"};
+
+  /// Direction of the beam to take into account TOF vs nominal IP8, can have
+  /// only values -1 or 1, or 0 to switch off the TOF and set time of
+  /// interaction to zero (default = 1, as for beam 1)
+  Gaudi::Property<int> m_zDir{"beamDirection", 0, "Direction of the beam, possible values: -1, 1 or 0"};
```
Comment (jlingema) [inline Generation/src/components/GaussSmearVertex.h]: You say the default is 1 but you set it to 0 below.

File: Generation/src/components/GaussSmearVertex.cpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  if (!sc.isSuccess()) return Error("Could not initialize normal random number generator");
+
+  release(randSvc);
+  return sc;
+}
+
+/// Smearing function
+StatusCode GaussSmearVertex::smearVertex(HepMC::GenEvent& theEvent) {
+  double dx, dy, dz, dt;
```
Comment (jlingema) [inline Generation/src/components/GaussSmearVertex.cpp]: Please initialize variables where you declare them. I.e. also declare them below?

File: Generation/src/components/GaussSmearVertex.cpp [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+  IRndmGenSvc* randSvc = svc<IRndmGenSvc>("RndmGenSvc", true);
+
+
+
+  sc = m_gaussDist.initialize(randSvc, Rndm::Gauss(0., 1));
+  if (sc.isFailure()) return sc;
+
+  std::string infoMsg = " applying TOF of interaction with ";
+  if (m_zDir == -1) {
```
Comment (jlingema) [inline Generation/src/components/GaussSmearVertex.cpp]: I don't see where this is used in the code except for these messages? Is the member necessary?

File: Generation/src/components/GaussSmearVertex.h (hunk 3 above)
Comment (vvolkl) [inline Generation/src/components/GaussSmearVertex.h]: As you mentioned, this property isn't actually used, so I just removed it. I did fix it in the original code, though (FlatSmearVertex).

File: Generation/src/components/GaussSmearVertex.cpp (hunk 5 above)
Comment (vvolkl) [inline Generation/src/components/GaussSmearVertex.cpp]: Removed it...

File: Generation/src/components/GaussSmearVertex.cpp (hunk 4 above)
Comment (vvolkl) [inline Generation/src/components/GaussSmearVertex.cpp]: Done!

**Pull request merged**